### ISSUE TITLE
we can't use BaseOpenStackService.Invocation to set the http request header parameters.

### ISSUE THREAD

**Issue by bboyHan** [openstack4j/openstack4j#88]
eg.   If i want to set http request header parameters about "X-Openstack-Nova-Api-Version", but the openstack4j can not use it properly.

// BaseOpenStackService.Invocation  method
public Invocation<R> header(String name, Object value) {
            req.header(name, value);
            return this;
 }

**Comment by olivergondza** [comment]
What is the use-case for the client code overriding the version of the API?

**Comment by bboyHan** [comment]
api link: https://docs.openstack.org/api-ref/compute/    

-> API Versions

It is mentioned in the document here that you need to use header to set “X-OpenStack-Nova-API-Version” to call the nova plugin api.

**Comment by olivergondza** [comment]
True, though my point is that this is something that openstack4j can and should do for the users. Whatever header have to pass in order for the API to work in a certain way is something the library can hide from the user. Making this configurable from the outside would just expose all this complexity to openstack4j users, and theoretically permitting them to request an API version that out model objects do not expect.

Do you have a user code that can benefit from changing/setting this header value? I would like to understand what id does and address that along the lines of the previous paragraph.
